src/pkcs11/framework-pkcs15.c

```diff
@@ -3730,6 +3730,10 @@ pkcs15_prkey_get_attribute(struct sc_pkcs11_session *session,
 		break;
     case CKA_ALWAYS_AUTHENTICATE:
 		check_attribute_buffer(attr, sizeof(CK_BBOOL));
+		if (fw_data->p15_card->opts.pin_cache_ignore_user_consent) {
+			*(CK_BBOOL*)attr->pValue = CK_FALSE;
+			break;
```

Comment:
putting the `break` here, doesn't seem so common. If you would just use an `else` branch, it looks more obvious, at least for me.

Reply:
No problem in changing this detail. I would be more interested it if works for the reporter use case and whether you are conceptually fine with this.

Reply:
I'm fine with it. A card, that *always* forces a re-validation of the PIN will be rendered useless if PIN-caching is disabled, but if the user sets this option manually, then I hope she knows what she's doing.